backend/prompt_studio/prompt_studio_index_manager_v2/models.py

```diff
@@ -0,0 +1,130 @@
+import json
+import logging
+import uuid
+
+from account_v2.models import User
+from django.db import models
+from django.db.models.signals import pre_delete
+from django.dispatch import receiver
+from prompt_studio.prompt_profile_manager_v2.models import ProfileManager
+from prompt_studio.prompt_studio_core_v2.prompt_ide_base_tool import PromptIdeBaseTool
+from prompt_studio.prompt_studio_document_manager_v2.models import DocumentManager
+from unstract.sdk.constants import LogLevel
+from unstract.sdk.vector_db import VectorDB
+from utils.models.base_model import BaseModel
+from utils.user_context import UserContext
+
+logger = logging.getLogger(__name__)
+
+
+class IndexManager(BaseModel):
+    """Model to store the index details."""
+
+    index_manager_id = models.UUIDField(
+        primary_key=True, default=uuid.uuid4, editable=False
+    )
+
+    document_manager = models.ForeignKey(
+        DocumentManager,
+        on_delete=models.CASCADE,
+        related_name="index_managers",
+        editable=False,
+        null=False,
+        blank=False,
+    )
+
+    profile_manager = models.ForeignKey(
+        ProfileManager,
+        on_delete=models.SET_NULL,
+        related_name="index_managers",
+        editable=False,
+        null=True,
+        blank=True,
+    )
```

Comment:
@muhammad-ali-e Do we need cascading delete here?

Reply:
@harini-venkataraman @gaya3-zipstack can you confirm this?

Reply:
Than clearing off Indexing done, when the profile manager is deleted, shouldn't we prevent its deletion when used elsewhere? Also, this opens up a bigger discussion as in, what will happen to the nodes already created in the vector DB.

Reply:
@muhammad-ali-e @harini-venkataraman 

1. Seems like the index manager is currently set to cascade delete when deleting the document manager instead, which in turn gets deleted if the custom tool is deleted.

2. Also we are keeping the history of doc IDs in `index_ids_history` field. On deleting a record for this table, we are triggering a vecor db cleanup of all the doc IDs in `index_ids_history`, in the `pre-delete` hook for index manager.

Considering the above, I believe it should be ok to keep this unchanged for now. We can revisit this later if needed. cc @tahierhussain